jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketCoreSession.java

```diff
@@ -797,9 +806,23 @@ public String toString()
 
     private class Flusher extends IteratingCallback
     {
+        private final FragmentingFlusher fragmentingFlusher;
+
         private final Queue<FrameEntry> queue = new ArrayDeque<>();
         FrameEntry entry;
 
+        public Flusher()
+        {
+            fragmentingFlusher = new FragmentingFlusher(WebSocketCoreSession.this)
```

Comment:
Why is it that this Flusher has-a FragmentingFlusher  rather than is-a FragmentingFlusher

If the relationship really is has-a, then I think it might be better just to add FragmentingExtention to the top of the extension stack